internal/clients/telegram/response/response.go

```diff
@@ -1,31 +1,162 @@
+	if err != nil {
+		return json.RawMessage{}, fmt.Errorf("network error: %w", err)
+	}
+
+	body, err = io.ReadAll(resp.Body)
+	if err != nil {
+		resp.Body.Close()
+
+		return json.RawMessage{}, fmt.Errorf("could not read response body %w", err)
+	}
+
+	resp.Body.Close()
+
+	var data struct {
+		Ok bool `json:"ok"`
+		APIError
+		Result json.RawMessage `json:"result,omitempty"`
+	}
+
+	if err = json.Unmarshal(body, &data); err != nil {
+		return data.Result, fmt.Errorf("parsing json response error: %w", err)
+	}
+
+	if !data.Ok {
+		return json.RawMessage{}, APIError{
+			ErrorCode:   data.ErrorCode,
+			Description: data.Description,
+			Parameters:  data.Parameters,
+		}
+	}
+
+	return data.Result, nil
+}
+
+func (r APIRequester) DoURLEncoded(ctx context.Context, endpoint string, params url.Values) (json.RawMessage, error) {
+	url := url.URL{
+		Scheme:   "https",
+		Host:     r.Host,
+		Path:     path.Join(r.BasePath, endpoint),
+		RawQuery: params.Encode(),
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url.String(), nil)
+	if err != nil {
+		// Delegates the correctness of the request to the one who is making it. If they can't ensure the request will
+		// be created, they should do it themselves.
+		return json.RawMessage{}, fmt.Errorf("while constructing URL encoded get request to /%s: %w", endpoint, err)
+	}
+
+	resp, err := r.Client.Do(req)
+	if err != nil {
+		return json.RawMessage{}, fmt.Errorf("network error: %w", err)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		resp.Body.Close()
+
+		return json.RawMessage{}, fmt.Errorf("could not read response body %w", err)
+	}
+
+	resp.Body.Close()
+
+	var data struct {
+		Ok bool `json:"ok"`
+		APIError
+		Result json.RawMessage `json:"result,omitempty"`
+	}
+
+	if err = json.Unmarshal(body, &data); err != nil {
+		return data.Result, fmt.Errorf("parsing json response error: %w", err)
+	}
+
+	if !data.Ok {
+		return json.RawMessage{}, APIError{
+			ErrorCode:   data.ErrorCode,
+			Description: data.Description,
+			Parameters:  data.Parameters,
+		}
+	}
+
+	return data.Result, nil
+}
+
 type BotAction interface {
-	URLEncode() (endpont string, params url.Values)
+	JSONEncode() (endpoint string, _ json.RawMessage, _ error)
 }
 
-// Noop returns empty list of bot actions.
-func Noop() []BotAction { return []BotAction{} }
+// Nothing returns an empty list of bot actions.
+func Nothing() []BotAction { return []BotAction{} }
+
+type SendMessage struct {
+	ChatID                ChatID                `json:"chat_id"`
+	Text                  string                `json:"text"`
+	ParseMode             option.Option[string] `json:"parse_mode,omitempty"`
+	DisableWebpagePreview bool                  `json:"disable_web_page_preview"`
+	ReplyMarkup           ReplyMarkupper        `json:"reply_markup,omitempty"`
+}
 
 // NewSendMessage creates a new NewSendMessage and sets the default parse mode to "html".
 func NewSendMessage(chatID ChatID, text string) SendMessage {
```

Comment:
Should be linted with exhauststruct so that if new fields are added defuaults for them are assigned.